File: takeItEasy-domain/src/main/java/com/sudosoo/takeiteasy/service/CouponServiceImpl.java
package com.sudosoo.takeiteasy.service;

import com.sudosoo.takeiteasy.dto.event.CreateEventRequestDto;
import com.sudosoo.takeiteasy.entity.Coupon;
import com.sudosoo.takeiteasy.repository.CouponRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import java.time.LocalDateTime;

@Service
@Transactional
@RequiredArgsConstructor
public class CouponServiceImpl implements CouponService {
    private final CouponRepository couponRepository;
    @Override
    public Coupon priceCouponCreate(CreateEventRequestDto requestDto) {
        Coupon coupon = Coupon.priceOf(requestDto);
        return couponRepository.save(coupon);
    }

    @Override
    public Coupon rateCouponCreate(CreateEventRequestDto requestDto) {
        Coupon coupon = Coupon.rateOf(requestDto);
        return couponRepository.save(coupon);
    }
}

File: takeItEasy-application/src/main/kotlin/com/sudosoo/takeItEasy/service/CouponServiceImpl.kt
package com.sudosoo.takeItEasy.service

import com.sudosoo.takeItEasy.dto.event.CreateEventRequestDto
import com.sudosoo.takeiteasy.dto.event.CreateEventRequestDto
import com.sudosoo.takeiteasy.entity.Coupon
import com.sudosoo.takeiteasy.repository.CouponRepository
import lombok.RequiredArgsConstructor
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional

@Service
@Transactional
@RequiredArgsConstructor
class CouponServiceImpl(val couponRepository: CouponRepository) : CouponService {
    override fun priceCouponCreate(requestDto: CreateEventRequestDto): Coupon {
        val coupon: Coupon = Coupon.priceOf(requestDto)
        return couponRepository.save<Coupon>(coupon)
    }

    override fun rateCouponCreate(requestDto: CreateEventRequestDto): Coupon {
        val coupon: Coupon = Coupon.rateOf(requestDto)
        return couponRepository.save<Coupon>(coupon)
    }
}
